drivers/soundwire/generic_bandwidth_allocation.c

```diff
@@ -360,7 +361,11 @@ static int sdw_compute_bus_params(struct sdw_bus *bus)
 				(bus->params.max_dr_freq >>  clk_buf[i]) :
 				clk_buf[i] * SDW_DOUBLE_RATE_FACTOR;
 
-		if (curr_dr_freq <= bus->params.bandwidth)
+		frame_int = mstr_prop->default_row * mstr_prop->default_col;
+		frame_freq = curr_dr_freq / frame_int;
+
+		if ((curr_dr_freq - (frame_freq * SDW_FRAME_CTRL_BITS)) <
```

Comment:
I think this is partially wrong, because we never use column0 for audio. so for a 50 row frame, this calculation would be off by 2 bits times the frame_freq. I think you need to use curr_dr_freq - default_row * frame_freq.

Reply:
This is copy from `sdw_select_row_col()`. Do we need to do the same change in `sdw_select_row_col()`, too?

Reply:
yeah, I think so. I don't think anyone can use column0 with 19.2 or 12.228 root frequencies. It's only possible for the 125-row frame but that's such a corner case.

But the formula could be simplifed to avoid a division btw:

curr_dr_freq - (curr_dr_freq / (sdw_rows[r] * sdw_cols[c]) * sdw_rows[r])

curr_d_freq - curr_dr_freq / sdw_cols[c] < bandwidth
curr_d_freq * (sdw_cols[c] - 1) < bandwidth * sdw_cols[c]